pkg/fqdn/name_manager_gc.go

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package fqdn
+
+import (
+	"context"
+	"net/netip"
+	"os"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/fqdn/matchpattern"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const DNSGCJobInterval = 1 * time.Minute
+
+var (
+	dnsGCJobName         = "dns-garbage-collector-job"
+	dnsGCControllerGroup = controller.NewGroup(dnsGCJobName)
+)
+
+// This implements some garbage cllection and cleanup functions for the NameManager
+
+// GC is a controller to cleanup TTL expired entries from the DNS policies.
+// dns-garbage-collector-job runs the logic to remove stale or undesired
```

Comment:
nit:
```suggestion
// This file implements some garbage collection and cleanup functions for the NameManager

// GC removes stale or undesired
```

(GC is no longer a controller after refactor, code does not match the function comment)